src/libraries/System.Net.Sockets/tests/FunctionalTests/UdpClientTest.cs

```diff
@@ -598,31 +628,71 @@ public async Task SendAsync_ReceiveAsync_Success(bool ipv4)
+				await sender.SendAsync(new ReadOnlyMemory<byte>(new byte[1]), new IPEndPoint(address, ((IPEndPoint)receiver.Client.LocalEndPoint).Port));
+				await AssertReceiveAsync(receiver);
+            }
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        [PlatformSpecific(TestPlatforms.Windows)] // "localhost" resolves to IPv4 & IPV6 on Windows, but may resolve to only one of those on Unix
+        public async Task SendAsync_ReceiveAsync_With_HostName_Success(bool ipv4)
+        {
+            IPAddress address = ipv4 ? IPAddress.Loopback : IPAddress.IPv6Loopback;
 
-                UdpReceiveResult result = await receiver.ReceiveAsync();
-                Assert.NotNull(result.RemoteEndPoint);
-                Assert.NotNull(result.Buffer);
-                Assert.InRange(result.Buffer.Length, 1, int.MaxValue);
+            using (var receiver = new UdpClient(new IPEndPoint(address, 0)))
+            using (var sender = new UdpClient(new IPEndPoint(address, 0)))
+            {
+                await sender.SendAsync(new byte[1], "localhost", ((IPEndPoint)receiver.Client.LocalEndPoint).Port);
+                await AssertReceiveAsync(receiver);
+
+                await sender.SendAsync(new ReadOnlyMemory<byte>(new byte[1]), "localhost", ((IPEndPoint)receiver.Client.LocalEndPoint).Port);
+                await AssertReceiveAsync(receiver);
+            }
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ReceiveAsync_Cancel_Throw(bool ipv4)
+        {
+            IPAddress address = ipv4 ? IPAddress.Loopback : IPAddress.IPv6Loopback;
+            
+            using (var receiver = new UdpClient(new IPEndPoint(address, 0)))
+            {
+                using (var timeoutCts = new CancellationTokenSource(1))
+                {
+                    await Assert.ThrowsAnyAsync<OperationCanceledException>(() => receiver.ReceiveAsync(timeoutCts.Token).AsTask());
+                }
             }
         }
 
         [Fact]
         [PlatformSpecific(TestPlatforms.Windows)] // "localhost" resolves to IPv4 & IPV6 on Windows, but may resolve to only one of those on Unix
-        [OuterLoop]
         public async Task SendAsync_ReceiveAsync_Connected_Success()
         {
             using (var receiver = new UdpClient("localhost", 0))
             using (var sender = new UdpClient("localhost", ((IPEndPoint)receiver.Client.LocalEndPoint).Port))
             {
                 await sender.SendAsync(new byte[1], 1);
+                await AssertReceiveAsync(receiver);
 
-                UdpReceiveResult result = await receiver.ReceiveAsync();
-                Assert.NotNull(result.RemoteEndPoint);
-                Assert.NotNull(result.Buffer);
-                Assert.InRange(result.Buffer.Length, 1, int.MaxValue);
+                await sender.SendAsync(new ReadOnlyMemory<byte>(new byte[1]));
+                await AssertReceiveAsync(receiver);
```

Comment:
Let's test the remaining overloads for the Connected behavior:

```suggestion
                await sender.SendAsync(new ReadOnlyMemory<byte>(new byte[1]));
                await AssertReceiveAsync(receiver);

                await sender.SendAsync(new ReadOnlyMemory<byte>(new byte[1]), null);
                await AssertReceiveAsync(receiver);
                
                await sender.SendAsync(new ReadOnlyMemory<byte>(new byte[1]), null, 0);
                await AssertReceiveAsync(receiver);
```

Reply:
Added.